File: Q4/python/abstract_factory/Factory.py
import classusers

class Factory:
    def __int__(self):
        pass
    
    def getObj(self, tag):
        obj = None
        if tag == 1:
            obj = classusers.Class1()
        elif tag == 2:
            obj = classusers.Class2()
        elif tag == 3:
            obj = classusers.Class3()
        elif tag == 4:
            obj = classusers.Class4()
        else:
            print('error')
        
        return obj
    
if __name__ == "__main__":
    f = Factory()
    obj = f.getObj(2)
    obj.getName()

File: Q4/php/abstract_factory/Factory.php
<?php
/**
 * 工程类，主要用来创建对象
 * 功能：输入相应的值, 返回对应的对象实例
 *
 */
require_once 'class/Class1.php';
require_once 'class/Class2.php';
require_once 'class/Class3.php';
require_once 'class/Class4.php';
class Factory
{

    public static function getInstance($tag)
    {
        switch ($tag) {
            case '1':
                return new Class1();
                break;
            case '2':
                return new Class2();
                break;
            case '3':
                return new Class3();
                break;
            case '4':
                return new Class4();
                break;
        }
    }
}

